sbom-utility-scripts/scripts/add-image-reference-script/README.md

```diff
@@ -0,0 +1,95 @@
+# Add image reference script
+
+The script aims to enrich the SBOM file with additional information about the images used in the build process.
+Based on a input SBOM type, the script updates certain fields with the image reference information. This is needed
+to provide a complete SBOM file that can be used for further analysis.
+
+## Usage
+
+```bash
+python add_image_reference.py \
+    --input-file ./input-sbom.json \
+    --output-path ./updated-sbom.json \
+    --image-url quay.io/foo/bar/ubi8:1.1 \
+    --image-digest sha256:011ff0cd8f34588d3eca86da97619f7baf99c8cc12e24cc3a7f337873c8d36cc \
+    --arch amd64
+```
+The script stores the updated SBOM in the output path provided.
+
+## List of updates
+### SPDX
+The script updates the following fields in the SPDX SBOM:
+- `packages` - the script adds a new package with the image reference information
+- `relationships` - the script adds a new relationship between the package and the image reference
+- `name` - the script adds the image reference as a name
+
+#### Example
+```json
+{
+  "name": "quay.io/foo/bar/ubi8@sha256:011ff0cd8f34588d3eca86da97619f7baf99c8cc12e24cc3a7f337873c8d36cc",
+  "packages": [
+    {
+      "SPDXID": "SPDXRef-image",
+      "name": "ubi8",
+      "versionInfo": "1.1",
+      "downloadLocation": "NOASSERTION",
+      "licenseConcluded": "NOASSERTION",
+      "supplier": "NOASSERTION",
+      "externalRefs": [
+        {
+          "referenceLocator": "pkg:oci/ubi8@sha256:011ff0cd8f34588d3eca86da97619f7baf99c8cc12e24cc3a7f337873c8d36cc?repository_url=quay.io/foo/bar/ubi8",
+          "referenceType": "purl",
+          "referenceCategory": "PACKAGE-MANAGER"
+        }
+      ],
+      "checksums": [
+        {
+          "algorithm": "SHA256",
+          "checksumValue": "011ff0cd8f34588d3eca86da97619f7baf99c8cc12e24cc3a7f337873c8d36cc"
+        }
+      ]
+    },
+  ],
+  "relationships": [
+    {
+      "spdxElementId": "SPDXRef-DOCUMENT",
+      "relationshipType": "DESCRIBES",
+      "relatedSpdxElement": "SPDXRef-image"
+    },
+  ]
+}
+
+```
+
+### CycloneDX
+- `components` - the script adds a new component with the image reference information
+- `metadata.component.purl` - the script adds the image reference as a purl
+- `metadata.component.name` - the script adds the image reference as a name
```

Comment:
IMO we don't need to add the component to `components`, it should be enough that it's in `metadata.component`

Reply:
s/don't need to/shouldn't/

My interpretation is that `components` is for dependencies, so the main component shouldn't be there

Reply:
This was requested by the TPA team since a missing reference under `components`  causes that SBOM in CycloneDX format is not analyzed correctly by the Guac and the SBOM is then ignored by the incident management process.

Reply:
That sounds like a bug in Guac :thinking:

Reply:
There are more use cases for having the image reference in a `components` field that can't be achieved by having the info in `metadata.component`. One of the limitations of the suggested field is that it can't have multiple purls.

Feel free to continue with this discussion here https://redhat-internal.slack.com/archives/C07MHR3MGLR/p1729693977566349.

For incident management and the CAMP project that is targeting the end of Q4, having the item in `components` seems to be a must-have requirement. The TPA is going to release a new 2.x version early next year, where this might be fixed. In the meantime, let's enable this and evaluate it later if you think it is an important detail.

Reply:
Ack, sounds reasonable